pkg/controllers/vdb/onlineupgrade_reconciler.go

```diff
@@ -512,6 +509,42 @@ func (r *OnlineUpgradeReconciler) postStartReplicationMsg(ctx context.Context) (
 	return r.postNextStatusMsg(ctx, startReplicationMsgInx)
 }
 
+// prepareReplication makes sure there is at least an Up node in the main cluster
+// and the sandbox, to perform replication.
+// Once we start using services for replication, we will check only the scs served by the services.
+func (r *OnlineUpgradeReconciler) prepareReplication(ctx context.Context) (ctrl.Result, error) {
+	// Skip if the VerticaReplicator has already been created.
```

Comment:
Add if we already promoted sandbox in the beginning of the function: 
if vmeta.GetOnlineUpgradeSandboxPromoted(r.VDB.Annotations) == vmeta.SandboxPromotedTrue {
		return ctrl.Result{}, nil
}